src/pointer.cr

```diff
@@ -78,7 +78,7 @@ struct Pointer(T)
   # ptr2.address # => 1238
   # ```
   def +(other : Int)
-    self + other.to_i64
+    self + LibC::PtrdiffT.new(other)
```

Comment:
You must hack the compiler for this to compile on 32bit platforms (`Pointer#+(Int64)` is a Crystal intrinsic). See below.